en/micro-integrator/docs/use-cases/examples/rest_api_examples/setting-query-params-outgoing-messages.md

```diff
@@ -47,13 +48,29 @@ There are two query parameters (customer name and ID) that must be set in the ou
 </api>                  
 ```
 
+## Reading a query parameter
```

Comment:
shall we divide this section into two sections and add the examples for them ( here both are discussed under one section )
```bash
uri-template="/{val1}/groups/{val2}/json?q1={v1}&q2={v2}"
```
1 - Reading Query Param
```xml
<property name="queryParam" expression="$url:q1"></property>
```
or

```xml
<property name="queryParam" expression="get-property('query.param.q1')"></property>
```

2- Reading Path Param
```xml
<property name="pathParam" expression="get-property('uri.var.val1')"></property>
```
You can refer this also for more details 

- https://medium.com/@prabushi/read-append-url-parameters-with-http-and-address-endpoints-in-wso2-enterprise-service-bus-1ae69184a940 
- http://jamsheert.blogspot.com/2014/06/read-path-param-in-wso2-esb-481.html
- https://asitha.github.io/cs/programming/wso2/esb/mediators/2017/08/14/get-query-params-with-wso2-esb/

Please get this verified as well

Reply:
Added this in.